python/notebooks/hive/HiveToBigquery_vertex_pipeline_pyspark.ipynb

```diff
@@ -0,0 +1,487 @@
+    "#### Build pipeline and run Dataproc Template on Vertex AI Pipelines to migrate HIVE tables to BigQuery\n",
+    "\n",
+    "For this, make sure that the service account used to run the notebook has the following roles:\n",
+    " - roles/dataproc.editor\n",
+    " - roles/dataproc.worker"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "095c13bf-c121-4465-88fc-04778ef35a36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "runtime_prop={}\n",
+    "runtime_prop['spark.hadoop.hive.metastore.uris']=HIVE_METASTORE\n",
+    "runtime_prop['mapreduce.fileoutputcommitter.marksuccessfuljobs'] = \"false\"\n",
+    "\n",
+    "def migrate_hive(EXECUTION_LIST):\n",
+    "    EXECUTION_LIST = EXECUTION_LIST\n",
+    "    aiplatform.init(project=PROJECT_ID, staging_bucket=GCS_STAGING_LOCATION)\n",
+    "\n",
+    "    @dsl.pipeline(\n",
+    "        name=\"hive-to-bq-pyspark\",\n",
+    "        description=\"Pipeline to migrate tables from hive to bq\",\n",
+    "    )\n",
+    "    def pipeline(\n",
+    "        project_id: str = PROJECT_ID,\n",
+    "        location: str = REGION,\n",
+    "        main_python_file_uri: str = MAIN_PYTHON_FILE,\n",
+    "        python_file_uris: list = PYTHON_FILE_URIS,\n",
+    "        jar_file_uris: list = JARS,\n",
+    "        subnetwork_uri: str = SUBNET\n",
+    "    ):\n",
+    "        for table in EXECUTION_LIST:\n",
+    "            BATCH_ID = \"hive2bq-{}-{}\".format(table,datetime.now().strftime(\"%s\"))\n",
+    "            TEMPLATE_SPARK_ARGS = [\n",
+    "                                    \"--template=HIVETOBIGQUERY\",\n",
+    "                                    \"--hive.bigquery.input.database={}\".format(INPUT_HIVE_DATABASE),\n",
+    "                                    \"--hive.bigquery.input.table={}\".format(table),\n",
+    "                                    \"--hive.bigquery.output.table={}\".format(table),\n",
+    "                                    \"--hive.bigquery.output.dataset={}\".format(OUTPUT_BIGQUERY_DATASET),\n",
+    "                                    \"--hive.bigquery.output.mode={}\".format(HIVE_OUTPUT_MODE),\n",
+    "                                    \"--hive.bigquery.temp.bucket.name={}\".format(TEMP_BUCKET)                                    ]\n",
+    "            #print(TEMPLATE_SPARK_ARGS)\n",
+    "            _ = DataprocPySparkBatchOp(\n",
+    "                project=project_id,\n",
+    "                location=location,\n",
+    "                batch_id=BATCH_ID,\n",
+    "                main_python_file_uri=main_python_file_uri,\n",
+    "                python_file_uris=python_file_uris,\n",
+    "                jar_file_uris=jar_file_uris,\n",
+    "                subnetwork_uri=subnetwork_uri,\n",
+    "                runtime_config_properties=runtime_prop,\n",
+    "                args=TEMPLATE_SPARK_ARGS,\n",
+    "            )\n",
+    "            time.sleep(3)\n",
```

Comment:
Yes we need it here otherwise multiple jobs will have same dataproc name (as we are using timestamp till seconds in the job name) and it will give error. changed it to time.sleep(1) to reduce overall time.

Reply:
BATCH_ID = "hive2bq-{}-{}".format(table,datetime.now().strftime("%s"))